RealmSwift/Tests/PerformanceTests.swift

```diff
@@ -0,0 +1,462 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import XCTest
+import RealmSwift
+
+#if !DEBUG
+
+private func createStringObjects(factor: Int) -> Realm {
+    let realm = Realm(inMemoryIdentifier: factor.description)
+    realm.write {
+        for _ in 0..<(1000 * factor) {
+            realm.create(SwiftStringObject.self, value: ["a"])
+            realm.create(SwiftStringObject.self, value: ["b"])
+        }
+    }
+    return realm
+}
+private var smallRealm: Realm!
+private var mediumRealm: Realm!
+private var largeRealm: Realm!
+
+class SwiftPerformanceTests: TestCase {
+    override class func setUp() {
```

Comment:
`createStringObjects()` is called in `setUp()` to create three realms with test objects, and strong references to those are stored in global variables to keep the in-memory realms alive. Each test which wants one of the pre-populated realms calls `getStringObjects()`, which re-opens the in-memory realm by creating a new Realm with the same identify (and actually gets the same RLMRealm from the cache), makes a copy of it, and returns a Realm referencing the copy.

The whole point is that `createStringObjects()` is only called once in the class `setUp()` function rather than in each test which needs a test realm, and the global variables are not accessed because they're used purely to keep the in-memory realms alive, with actual access to the data being done by creating new Realms with the same identifier.